mne/io/eeglab/eeglab.py

```diff
@@ -55,34 +55,47 @@ def _to_loc(ll):
     if isinstance(ll, (int, float)) or len(ll) > 0:
         return ll
     else:
-        return 0.
+        return np.nan
 
 
 def _get_info(eeg, montage, eog=()):
     """Get measurement info."""
+    from scipy import io
     info = _empty_info(sfreq=eeg.srate)
+    update_ch_names = True
 
     # add the ch_names and info['chs'][idx]['loc']
     path = None
     if not isinstance(eeg.chanlocs, np.ndarray) and eeg.nbchan == 1:
             eeg.chanlocs = [eeg.chanlocs]
 
     if len(eeg.chanlocs) > 0:
-        ch_names, pos = list(), list()
+        pos_fields = ['X', 'Y', 'Z']
+        if (isinstance(eeg.chanlocs, np.ndarray) and not
+            isinstance(eeg.chanlocs[0], io.matlab.mio5_params.mat_struct)):
```

Comment:
It seems `mat_struct` has been in scipy already in may of 2009 (although I am not sure if in was present in the same place):
http://scipy-user.10969.n7.nabble.com/io-loadmat-and-mat-struct-object-td11508.html
that would mean that scipy >= 0.7 should have it - so we are safe :)